src/ucp/rma/flush.c

```diff
@@ -15,17 +15,43 @@
 #include "rma.inl"
 
 
-static void ucp_ep_flush_error(ucp_request_t *req, ucs_status_t status)
+static void
+ucp_ep_flush_request_change_uct_cmp(ucp_request_t *req, int diff,
+                                    ucp_lane_map_t new_started_lanes)
+{
+    ucs_assertv((req->send.state.uct_comp.count + diff) >= 0,
+                "req %p: comp %p count %d, diff %d", req,
+                &req->send.state.uct_comp, req->send.state.uct_comp.count,
+                diff);
+    ucs_assertv((req->send.flush.started_lanes & new_started_lanes) == 0,
+                "req %p: started_lanes 0x%x, new_started_lanes 0x%x", req,
+                req->send.flush.started_lanes, new_started_lanes);
+
+    req->send.state.uct_comp.count += diff;
+    req->send.flush.started_lanes  |= new_started_lanes;
+
+    ucp_trace_req(req,
+                  "ep %p flush changed completion count %d by %d, "
+                  "started lanes 0x%x by 0x%x", req->send.ep,
+                  req->send.state.uct_comp.count, diff,
```

Comment:
but `req->send.state.uct_comp.count` contains a result after applying `diff`. maybe rephrase it:
```
"ep %p flush changed completion count by %d - result %d, "
```